src/main/java/TaskListener.java

```diff
@@ -0,0 +1,191 @@
+            String byArgumentValue = arguments.substring(byCommandIndex)
+                    .substring("/by".length()).trim();
+
+            if (name.isEmpty()) {
+                printLines("Please provide name argument");
+                return;
+            }
+
+            if (byArgumentValue.isEmpty()) {
+                printLines("Please provide /by argument");
+                return;
+            }
+            Deadline deadline = new Deadline(name, false, byArgumentValue);
+            this.taskList.addTask(deadline);
+            printAddedTask(deadline);
+        }
+    }
+
+    private void eventCommandHandler(String arguments) {
+        {
+            int fromCommandIndex = arguments.indexOf("/from");
+            int toCommandIndex = arguments.indexOf("/to");
+
+            if (fromCommandIndex < 0) {
+                printLines("Please specify the /from command");
+                return;
+            }
+            if (toCommandIndex < 0) {
+                printLines("Please specify the /to command");
+                return;
+            }
+
+            String name = arguments.substring(0, fromCommandIndex).trim();
+
+            String fromArgumentValue = arguments.substring(fromCommandIndex, toCommandIndex)
+                    .substring("/from".length()).trim();
+
+            String toArgumentValue = arguments.substring(toCommandIndex)
+                    .substring("/to".length()).trim();
+
+            if (name.isEmpty()) {
+                printLines("Please provide the name argument");
+                return;
+            }
+
+            if (fromArgumentValue.isEmpty()) {
+                printLines("Please provide the /from argument");
+                return;
+            }
+
+            if (toArgumentValue.isEmpty()) {
+                printLines("Please provide the /to argument");
+                return;
+            }
+
+            Event event = new Event(name, false, fromArgumentValue, toArgumentValue);
+            this.taskList.addTask(event);
+            printAddedTask(event);
+        }
+    }
+
+    private void modifyDoneValueHandler(boolean isDone, int taskIndex) {
+        String successMessage = isDone ? "Nice! I've marked this task as done:"
+                : "OK, I've marked this task as not done yet:";
+
+        // Invalid index value
+        if (taskIndex < 0 || taskIndex >= this.taskList.getTasks().length) {
+            printLines("No task at such index!");
+            return;
+        }
+
+        Task task = this.taskList.getTasks()[taskIndex];
+        task.setIsDone(isDone);
+        printLines(successMessage, task.toString());
+    }
+
+    public void listen() {
+        Scanner in = new Scanner(System.in);
+
+        String line = in.nextLine();
+        String[] inputArray = line.split("\\s+", 2);
+
+        String command = inputArray[0];
+        String arguments = inputArray.length > 1 ? inputArray[1] : "";
+
+        switch (command) {
```

Comment:
switch statement should have no indentation for case clauses. For example:
switch (command) {
case "add":
<4 spaces>addCommandHandler(arguments);
<4 spaces>break;
......
}

Reply:
Thanks for a relevant suggestion! Will fix that indeed